recipes/kmod/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+        "with_zstd": [True, False],
+        "with_xz": [True, False],
+        "with_zlib": [True, False],
+        "with_openssl": [True, False],
+        "experimental": [True, False],
+        "logging": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "with_zstd": True,
+        "with_xz": True,
+        "with_zlib": True,
+        "with_openssl": True,
+        "experimental": False,
+        "logging": False,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def generate(self):
+        yes_no = lambda v: "yes" if v else "no"
+        tc = AutotoolsToolchain(self)
+        tc.default_configure_install_args = True
+        tc.configure_args.append("--with-zstd=%s" % yes_no(self.options.with_zstd))
+        tc.configure_args.append("--with-xz=%s" % yes_no(self.options.with_xz))
+        tc.configure_args.append("--with-zlib=%s" % yes_no(self.options.with_zlib))
+        tc.configure_args.append("--with-openssl=%s" % yes_no(self.options.with_openssl))
+        tc.configure_args.append("--enable-experimental=%s" % yes_no(self.options.experimental))
+        tc.configure_args.append("--enable-logging=%s" % yes_no(self.options.logging))
+        tc.configure_args.append("--enable-debug=%s" % yes_no(self.settings.build_type == "Debug"))
+        tc.configure_args.append("--enable-tools=no")
+        tc.configure_args.append("--enable-manpages=no")
+        tc.configure_args.append("--enable-test-modules=no")
+        tc.configure_args.append("--enable-python=no")
+        tc.configure_args.append("--enable-coverage=no")
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = AutotoolsDeps(self)
+        tc.environment.define("PKG_CONFIG_PATH", self.source_folder)  # is it really needed?
+        tc.environment.define("LIBS", "-lpthread")  # FIXME: https://github.com/conan-io/conan/issues/10640 & https://github.com/conan-io/conan/issues/10341
+        tc.environment.define("libzstd_LIBS", "-lzstd")
+        tc.generate()
+
+    def requirements(self):
+        if self.options.with_zstd:
+            self.requires('zstd/1.5.2')
+        if self.options.with_xz:
+            self.requires('xz_utils/5.2.5')
+        if self.options.with_zlib:
+            self.requires('zlib/1.2.11')
+        if self.options.with_openssl:
+            self.requires('openssl/3.0.1')
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("kmod is Linux-only!")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        self.run(os.path.join(self._source_subfolder, "autogen.sh"))
```

Comment:
```suggestion
        self.run(os.path.join(self._source_subfolder, "autogen.sh"))
```

libtool & pkgconf should be added to build requirements. `autogen.sh` call autoreconf... but also gtkdocize which may not be available.

Why not just call autoreconf manually to have full control of what happen?

Reply:
in their [readme](https://github.com/kmod-project/kmod#hacking), they say to launch `autogen.sh`, I just followed their build instructions.
`gtkdocize` seems to be fully optional (I don't build any docs).
`libtool`/`pkgconf` seems to be already available in docker images, but we can add explicit build_requirements for the completeness, as it's harmless (and will benefit users who will build on pure environment).

Reply:
If you run autogen.sh, gtkdocize is not optional anymore. My advice is to never run any autogen.sh script but ensure to run our own autoreconf. libtool & pkgconf must be added to build requirements because you don't know whether folks have libtool or pkg-config installed.

Reply:
as I can see, it's fully optional, e.g. `autogen.sh` prints:
```
source_subfolder/autogen.sh: 9: gtkdocize: not found
```
and then build continues and finishes successfully.
never mind, I've added explicit call to `autoreconf`

Reply:
I've also added libtool/pkgconf for completeness. it's the old good discussion on what to add to the build_requires, and what to assume as a ground base (or add to the profile, but not to the recipe). e.g. we assume that `cmake` should be already available on the developer's system, as well as `binutils` and `coreutils`, and the compiler itself, of course. it really only depends on where do you want to mark a border, and I am personally fine with libtool/pkgconf as build requirements.

Reply:
I've also added libtool/pkgconf for completeness. it's the old good discussion on what to add to the build_requires, and what to assume as a ground base (or add to the profile, but not to the recipe). e.g. we assume that `cmake` should be already available on the developer's system, as well as `binutils` and `coreutils`, and the compiler itself, of course. it really only depends on where do you want to mark a border, and I am personally fine with libtool/pkgconf as build requirements.

Reply:
I've also added libtool/pkgconf for completeness. it's the old good discussion on what to add to the build_requires, and what to assume as a ground base (or add to the profile, but not to the recipe). e.g. we assume that `cmake` should be already available on the developer's system, as well as `binutils` and `coreutils`, and the compiler itself, of course. it really only depends on where do you want to mark a border, and I am personally fine with libtool/pkgconf as build requirements.